packages/auth/src/grant/routes.ts

```diff
@@ -1,17 +1,17 @@
 import { AppContext } from '../app'
-import { GrantService } from './service'
+import { GrantService, GrantRequest } from './service'
 import { ClientService } from '../client/service'
 import { BaseService } from '../shared/baseService'
+import { isAccessRequest } from '../access/types'
 
 interface ServiceDependencies extends BaseService {
   grantService: GrantService
   clientService: ClientService
 }
 
 export interface GrantRoutes {
-  auth: {
-    post(ctx: AppContext): Promise<void>
-  }
+  validateGrantRequest(grantRequest: GrantRequest): boolean
```

Comment:
This might not need to be exposed by `GrantRoutes`.
It could instead be tested within the `create` tests (checking that it returns `400`)